### PR TITLE
added type hints to CTRL pytorch 

### PR DESCRIPTION
**Pull request by anmolsjoshi**
# What does this PR do?

I added type annotations for CTRL (PT) as described in #16059

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger

### CODE REVIEW

File: src/transformers/models/ctrl/modeling_ctrl.py [hunk 1]
```diff
@@ -359,18 +359,18 @@ def _prune_heads(self, heads_to_prune):
...
-        position_ids=None,
-        head_mask=None,
-        inputs_embeds=None,
-        use_cache=None,
-        output_attentions=None,
-        output_hidden_states=None,
-        return_dict=None,
-    ):
+        input_ids: torch.LongTensor = None,
+        past_key_values: Tuple[Tuple[torch.FloatTensor]] = None,
```
Comment (sgugger) [inline src/transformers/models/ctrl/modeling_ctrl.py:363]: Those two are optional as well (this applies to all the other forward pass you touched).

**Review by sgugger — COMMENTED** [review]
Thanks for working on this model! More of the inputs are optional, if you can fix it.

File: src/transformers/models/ctrl/modeling_ctrl.py (hunk 1 above)
Comment (anmolsjoshi) [reply src/transformers/models/ctrl/modeling_ctrl.py]: @sgugger thanks for your comment. Incorporated.

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Pull request merged**